storage/redis/adapter.go

```diff
@@ -0,0 +1,101 @@
+package redis
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	r "gopkg.in/redis.v3"
+)
+
+type adapter struct {
+	bufferSize  int
+	redisClient *r.Client
+}
+
+// NewStorageAdapter returns a pointer to a new instance of a redis-based storage.Adapter.
+func NewStorageAdapter(bufferSize int) (*adapter, error) {
```

Comment:
We've actually had this exact conversation before (I think) in reference to one of the other storage adapters. Anyone using this should be coding to the `storage.Adapater` interface that _is_ exported (it also exports all its functions). Also, fwiw, "anyone" is really just other bits of logger code. This isn't any sort of reusable library.